lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4159,7 +4159,11 @@ bool LLVMToSPIRVBase::transGlobalVariables() {
       continue;
     } else if (MDNode *IO = ((*I).getMetadata("io_pipe_id")))
       transGlobalIOPipeStorage(&(*I), IO);
-    else if (!transValue(&(*I), nullptr))
+    // As global variables define a pointer to their "content" type, we should
+    // translate here only pointer without declaration even it is a function
+    // pointer.
+    else if (!transValue(&(*I), nullptr, true /*CreateForward - default value*/,
+                         FuncTransMode::Pointer))
```

Comment:
... which means this change would be unnecessary. It feels wrong to me that the code in the writer responsible for causing the global value's definition to get emitted is telling the value emission code to *not* emit definitions (even if that mode only affects functions, not global values in general).

Reply:
Thanks! You are right, it'd be better to call the function with the needed options directly rather than passing through.
I'll upload the fixing change shortly.